sysboundary/ionosphere.cpp

```diff
@@ -1288,68 +1288,105 @@ namespace SBC {
          }
       }
 
-      uint32_t nearestNode;
-      for (uint toto=0; toto<2; toto++) {
-         // Determine the nearest ionosphere node to this point.
-         nearestNode = findNodeAtCoordinates(x);
-
+      // Determine the nearest ionosphere node to this point.
+      uint32_t nearestNode = findNodeAtCoordinates(x);
+      int32_t elementIndex = nodes[nearestNode].touchingElements[0];
+      int32_t oldElementIndex;
+      
+      std::unordered_set<int32_t> elementHistory;
+      bool override=false;
+      
+      for (uint toto=0; toto<15; toto++) {
```

Comment:
This hardcoded number 15 is a little weird here.

Also: if the loop has run through, should it actually continue to return an empty coupling, or just couple to whatever the closest node was?

Reply:
Could be something like MAX_TOUCHING_ELEMENTS? Eventually it should exit anyway, but as long as I'm debugging I don't put a `while(true)`. Shall I add a warning to make this nicer eventually?